### DISCUSSION TITLE
Protect the house batteries, while charging a car

### DISCUSSION THREAD

**Discussion by booti101** [nathanmarlor/foxess_em#319] [General]
Hi @nathanmarlor 
Thanks for this great piece of software. Been using it for a while now and loving what it does, but now i have moved on to my next step (an EV car) and wondered if you could help with a configuration question please.

I have an Eco7 tariff, so my off peak is set to 01:30 to 08:30 in the integration.
I see that you turn on this time period every day, then turn it off if there is sufficient battery to last the night until PV is available (with a minimum SoC set, just in case)

Last night i wanted to charge the EV during off peak. I set the scheduled charge time as 02:00 to 07:00 on my Zappi and set the 2nd 'force charge' period on the Fox Cloud to 01:50 to 07:15. This was showing the correct times in the Modbus Integration and the 'Enable Force Charge Period' was enabled.

I have come back this morning to find the car has charged, but it pulled 4.5kW from the battery as well.
It seems at 01:26 the integration turned off the 'Enable Force Charge Period' setting (it did period 1 at 01:30) The battery then discharged to the lowest SoC (27%) which i assume is what the integration had the minimum set to.

How do i prevent the integration from turning off period 2 please?

**Comment by nathanmarlor** [comment]
Hi @booti101 - thanks! Any settings you change outside of the off-peak period are reset right before the off-peak period begins. 

Thinking out loud - this could be solved with a 'hold charge' toggle, which prevents any discharge during the off-peak period. I'll have a look at the code when I get some free time and see how difficult it would be to implement :)

**Reply by booti101** [reply]
Hi @nathanmarlor 
Thanks. Yes if you could please, as we obviously dont want to be staying up to 2am to set a period so the house batteries arent used for the car.
Not sure what the implications would be of only restting the period 1 times maybe?
